### ISSUE TITLE
single infer 在自己的图片上输出 text 是 null 或者 None

### ISSUE THREAD

**Issue by blackbean001** [XPixelGroup/DepictQA#23]
你好，

我在使用单一的图片推理的时候发现输出的都是 null， 好像是  outputs = self.llm.generate(**inputs_generate) 输出的就是有问题的了。下面是我的 setting。可以麻烦帮忙看一下是否是这些设置哪里有问题么。好像使用的 ckpt.pt 有影响，但是使用 “DQ-495K + KonIQ + SPAQ“ 和 "DQ-495K + KonIQ + SPAQ" 都不能正常输出 text（分别是 null 和 None）。

#### 1. infer.sh
#!/bin/bash

CUDA_VISIBLE_DEVICES=1 \
python src/infer.py \
    --meta_path ./demo/meta.json \
    --dataset_name demo_jason \
    --task_name quality_single_A_noref \
    --batch_size 1 \
    --cfg ./demo/config.yaml

#### 2. config.yaml
seed: 131

data:
  root_dir: /home/jason/DepictQA
  num_workers: 2

  infer:  # set args to overwrite cfg.data.infer
    meta_path: null
    dataset_name: null
    task_name: null
    batch_size: null

model:
  vision_encoder_path: ../ModelZoo/CLIP/clip/ViT-L-14.pt
  vision_feature_type: local  # [local, global]
  llm_path: ../ModelZoo/LLM/vicuna/vicuna-7b-v1.5/
  llm_conv_type: vicuna_v1  # conv -> conversation
  lora:  # lora hyper-parameters
    lora_r: 16
    lora_alpha: 16
    lora_dropout: 0.1
    lora_target_modules: ['q_proj', 'k_proj', 'v_proj', 'o_proj']
  delta_path: /home/jason/DepictQA/experiments/a_specific_experiment_directory/ckpt/ckpt.pt  # load trained params, for resume / infer
  unique_tag: true

train:
  epochs: 1
  warmup_rate: 0.1
  log_step: 5
  save_freq_step: 5000  # can be set null
  save_freq_epoch: 1
  save_dir: ./ckpt/
  log_dir: ./log/
  max_tokens: 512  # query + answer (no system & vision)

infer:
  output_prob_id: false
  output_confidence: true
  sentence_model: ../ModelZoo/SentenceTransformers/all-MiniLM-L6-v2
  answer_dir: ./answers/
  temperature: 0.0  # 0.0 -> do_sample = False
  top_p: 0.9
  max_new_tokens: 400  # only answer

vision_preprocess:
  patch_size: 14
  resize: 224
  max_size: 672
  crop_ratio: [0.7, 1.]
  keep_ratio: true

deepspeed:
  train_micro_batch_size_per_gpu: 1
  gradient_accumulation_steps: 16
  gradient_clipping: 1.0
  steps_per_print: 1

  zero_optimization:
    allgather_bucket_size: 500000000
    allgather_partitions: true
    contiguous_gradients: true
    offload_optimizer:
      device: cpu
    stage: 1

  optimizer:
    type: Adam
    params:
      betas:
      - 0.9
      - 0.95
      eps: 1.0e-08
      lr: 0.0002
      weight_decay: 0.001

  scheduler:
    type: WarmupDecayLR
    params:
      total_num_steps: 20000
      warmup_max_lr: 0.0002
      warmup_min_lr: 1.0e-08
      warmup_num_steps: 10

  fp16:
    enabled: true
    min_loss_scale: 1
    opt_level: O2

  bf16:
    enable: false

  activation_checkpointing:
    partition_activations: true
    cpu_checkpointing: true
    contiguous_memory_optimization: false
    number_checkpoints: null
    synchronize_checkpoint_boundary: false
    profile: false

### 3. meta.json
[
    {
        "distortion_classes": [
            "saturate_strengthen",
            "blur"
        ],
        "distortion_names": [
            "saturate_strengthen_HSV",
            "blur_gaussian"
        ],
        "severities": [
            3,
            2
        ],
        "id": "sky-79124_0",
        "image_ref": null,
        "image_A": "/home/jason/DiffBIR/results/demo_superres/00048_unwarp.png",
        "image_B": null,
        "task_type": "quality_single_A_noref",
        "query": "Identify the chief TWO degradations in the evaluated image. Answer the question using a single word or phrase.",
    }
]

**Comment by blackbean001** [comment]
<img width="728" height="625" alt="Image" src="https://github.com/user-attachments/assets/81612502-7830-4beb-b890-8a3ad16dd2e9" />

**Comment by blackbean001** [comment]
<img width="600" height="114" alt="Image" src="https://github.com/user-attachments/assets/3d2bf655-49a0-445c-ac1c-18d540a5479e" />

**Comment by blackbean001** [comment]
输入 embedding 看着正常，但是经过 self.llm.generate(**inputs_generate) 后输出异常

<img width="695" height="652" alt="Image" src="https://github.com/user-attachments/assets/022f39bb-d3a9-4ae2-95e7-d95f91f7b78c" />

**Comment by blackbean001** [comment]
多次尝试，似乎与 ckpt 选择，图片以及query均有关系。
close

**Comment by zhiyuanyou** [comment]
Hello，感谢你的关注和兴趣！能告知一下是哪个ckpt，哪个query吗？我记录一下，之后出现相同的问题可以解答～

**Comment by blackbean01** [comment]
> Hello，感谢你的关注和兴趣！能告知一下是哪个ckpt，哪个query吗？我记录一下，之后出现相同的问题可以解答～

用的是README中第一个包含extractor的ckpt（DQ-495K + KonIQ + SPAQ）, prompt 是 single 和 compare 好像都可能有 null

**Comment by blackbean01** [comment]
> Hello，感谢你的关注和兴趣！能告知一下是哪个ckpt，哪个query吗？我记录一下，之后出现相同的问题可以解答～

不好意思可能再麻烦问一下，我这里目前使用六块 A5000 训练的时候也会出现 oom。我理解目前使用的是 data parallel 每个卡还是会保存一份完整的模型参数，所以想要少用显存的话可能换成 tp 或者 pp，或者减小 extractor 的 dim 是这样么。或者请问有没有其他的降低显存的经验呢，

谢谢解答！
